### DISCUSSION TITLE
ROS question

### DISCUSSION THREAD

**Discussion by johncook1212** [iNavFlight/inav#6144] [Q&A]
Hi,

I hope this is the right place to ask...
Is there a separation between ROS and INAV/betaflight firmware? 
Looking at src folder I see the code related to sensors and drivers. Parallel, looking at ArduCopter's docs - there is a separation between different ROS versions (NuttX, ChibiOS etc.)
I would be happy to understand the line between those?
Thanks
John

[a comment by an automated account is not shown]

**Comment by johncook1212** [comment]
anyone? appreciate a "keyword answer" regarding the "border" between those (ROS, INAV)
Best,
John

**Comment by amfern** [comment]
INav doesn't work with ros2, i made a PR to try and rectify it https://github.com/iNavFlight/inav/pull/7061
